lib/handler/connect.c

```diff
@@ -116,40 +116,91 @@ struct st_connect_generator_t {
     };
 };
 
+static h2o_iovec_t get_proxy_status_identity(struct st_connect_generator_t *self)
+{
+    h2o_iovec_t identity = self->src_req->conn->ctx->globalconf->proxy_status_identity;
+    if (identity.base == NULL)
+        identity = h2o_iovec_init(H2O_STRLIT("h2o"));
+    return identity;
+}
+
+static h2o_iovec_t get_dest_addr_str(struct st_connect_generator_t *self, char *dest_addr_strbuf,
+                                     const struct st_server_address_t **addr_out)
+{
+    const struct st_server_address_t *addr = NULL;
+    h2o_iovec_t dest_addr_str = h2o_iovec_init(NULL, 0);
+    if (self->server_addresses.used > 0 && self->server_addresses.used <= self->server_addresses.size) {
```

Comment:
I think that we need something for when the function is called from `record_error` because that can happen before there are any addresses.  Would you be fine with just `if (self->server_addresses.used > 0)`?  That should detect the case when there are no IP addresses yet correctly.

Reply:
Ah, thanks for pointing out the error case. I see it now.